plugins/woocommerce-blocks/assets/js/data/checkout/utils.ts

```diff
@@ -131,8 +137,11 @@ export const runCheckoutSuccessObservers = ( {
 	dispatch: ActionCreatorsOf< ConfigOf< typeof checkoutStore > >;
 	createErrorNotice: typeof originalCreateErrorNotice;
 } ) => {
-	let successResponse = null as null | Record< string, unknown >;
-	let errorResponse = null as null | Record< string, unknown >;
+	let successResponse = null as null | ObserverSuccessResponse;
```

Comment:
Small TS update as success and error responses should be one of the observer types, which now extends `Record< string, unknown >`.